x-pack/plugins/viz_editor/common/query_types.ts

```diff
@@ -8,28 +8,20 @@
 // Select clause
 // ----------------------------------------
 
-export type SelectOperator =
-  | 'column'
-  | 'count'
-  | 'cardinality'
-  | 'date_histogram'
-  | 'sum'
-  | 'avg'
-  | 'terms'
-  | 'window';
+export type SelectOperator = SelectOperation['operator'];
 
-export interface Aliasable {
+export interface BasicOperation {
   operator: SelectOperator;
-  alias: string;
+  id: string;
 }
 
 export interface Field {
   field: string;
 }
 
-export type FieldOperation = Aliasable & {
+export interface FieldOperation extends BasicOperation {
   argument: Field;
-};
+}
```

Comment:
Not a required change, but could you combine `BasicOperation` and `FieldOperation`, like:

```
export interface BasicFieldOperation {
  operator: SelectOperator;
  id: string;
  argument?: Field;
}
```

the naming is not great, but consistency here seems useful

Reply:
In the current type hierarchy we have BasicOperation -> CountOperation and BasicOperation ->  FieldOperation -> (TermsOperation | AvgOperation | ...). Not sure whether having a BasicFieldOperation interface which is used for things which not even work on a field (currently just count, but there will be others) is an improvement over the current state, but happy to discuss. However I think polishing such details can be postponed to later.